src/ucp/proto/proto_select.c

```diff
@@ -892,3 +893,77 @@ void ucp_proto_select_param_str(const ucp_proto_select_param_t *select_param,
         ucs_string_buffer_appendf(strb, " and fast completion");
     }
 }
+
+void ucp_proto_select_short_disable(ucp_proto_select_short_t *proto_short)
+{
+    proto_short->max_length_unknown_mem = -1;
+    proto_short->max_length_host_mem    = -1;
+    proto_short->lane                   = UCP_NULL_LANE;
+    proto_short->rkey_index             = UCP_NULL_RESOURCE;
+}
+
+void
+ucp_proto_select_short_init(ucp_worker_h worker, ucp_proto_select_t *proto_select,
+                            ucp_worker_cfg_index_t ep_cfg_index,
+                            ucp_worker_cfg_index_t rkey_cfg_index,
+                            ucp_operation_id_t op_id, uint32_t op_attr_mask,
+                            unsigned proto_flags,
+                            ucp_proto_select_short_t *proto_short)
+{
+    ucp_context_h context    = worker->context;
+    const ucp_proto_t *proto = NULL;
+    const ucp_proto_threshold_elem_t *thresh;
+    ucp_proto_select_param_t select_param;
+    const ucp_proto_single_priv_t *spriv;
+    uint32_t op_attr;
+
+    ucs_for_each_submask(op_attr, op_attr_mask) {
```

Comment:
why is this needed? Can you please give an example of set of flags?

Reply:
added comment